test/suites/integration/ami_test.go

```diff
@@ -314,3 +330,19 @@ func getInstanceAttribute(nodeName string, attribute string) *ec2.DescribeInstan
 	Expect(err).ToNot(HaveOccurred())
 	return instanceAttribute
 }
+
+func EventuallyExpectAMIs(provider *v1alpha1.AWSNodeTemplate, amiID string) {
+	Eventually(func(g Gomega) {
+		var ant v1alpha1.AWSNodeTemplate
+		g.Expect(env.Client.Get(env, client.ObjectKeyFromObject(provider), &ant)).To(Succeed())
+		amiIDsInStatus := lo.Map(ant.Status.AMIs, func(ami v1alpha1.AMI, _ int) string {
+			return ami.ID
+		})
+		g.Expect(amiIDsInStatus).To(Equal([]string{amiID}))
```

Comment:
Why not ensure the entire status? Also, if you are checking for existence in a list, you can use Ginkgo's `Contains`